src/main/java/lotto/sevice/Calculator.java

```diff
@@ -0,0 +1,42 @@
+package lotto.sevice;
+
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lotto.domain.Lotto;
+import lotto.domain.PurchaseAmount;
+import lotto.domain.UserInputNumbers;
+import lotto.domain.WinningCheck;
+
+public class Calculator {
+
+  private static final int PERCENTAGE_FACTOR = 100;
+
+  public Map<WinningCheck, Integer> calculateWinningResult(List<Lotto> lottos,
+      UserInputNumbers receivedLotto) {
+    Map<WinningCheck, Integer> result = new HashMap<>(); // WinningCheck 객체 정수를 키와 값으로 Map에 저장
+    for (Lotto lotto : lottos) {
+      int matchCount = lotto.getMatchCount(receivedLotto);
+      boolean bonusMatch = lotto.isBonusMatch(receivedLotto); // receivedLotto는 bonus를 포함하고있음
+
+      WinningCheck prize = WinningCheck.getPrize(matchCount, bonusMatch);
+      result.merge(prize, 1, Integer::sum);
+    }
+    return result;
+  }
+
+  public double calculateProfitRate(PurchaseAmount purchaseAmount,
+      Map<WinningCheck, Integer> result) {
+    double totalPrize = 0;
+    for (Map.Entry<WinningCheck, Integer> entry : result.entrySet()) {
+      WinningCheck prize = entry.getKey();
+      int count = entry.getValue();
+      totalPrize += prize.getPrizeAmount() * count;
+    }
+
```

Comment:
```suggestion
double totalPrize = (double) result.entrySet().stream()
.mapToInt(entry -> entry.getKey().getPrizeAmount() * entry.getValue())
.sum();

```

사실 어느게 깔끔한진 모르겠지만 이렇게 스트림으로 지역변수 선언과 할당을 동시에 할 수도 있습니다. ㅋㅋㅋ

Reply:
저는 스트림이 더 좋은 코드라 생각합니다. 명확하네요!